src/wallet/wallet2.h

```diff
@@ -1201,7 +1203,18 @@ namespace tools
       a & m_scanned_pool_txs[1];
       if (ver < 20)
         return;
-      a & m_subaddresses.parent();
+      if (ver < 31)
+      {
+        using pair_type = std::pair<crypto::public_key, cryptonote::subaddress_index>;
+        std::unordered_map<crypto::public_key, cryptonote::subaddress_index> subaddresses;
+        a & subaddresses;
+        std::vector<pair_type> vector;
+        vector.reserve(subaddresses.size());
+        for (const auto &e: subaddresses)
+          vector.push_back(e);
```

Comment:
Can't you do: `std::vector<pair_type> vector{ subaddresses.begin(), subaddresses.end() };` ?